entproto/cmd/protoc-gen-entgrpc/codec.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019-present Facebook
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+
+	"entgo.io/contrib/entproto"
+	"google.golang.org/protobuf/compiler/protogen"
+	dpb "google.golang.org/protobuf/types/descriptorpb"
+)
+
+type fieldCodec struct {
```

Comment:
I think `convert` fits better here, because we don't encode/decode anything (right?), it's converting types and (un)wrap with proto-types if needed.